Compile bug: ggml-virtgpu/ggml-backend-device.cpp fails to compile on Fedora44/g++ 16.1.1

### Git commit

$ git rev-parse HEAD
deab41ec68d81324117017a5dd10452920c5c915

### Operating systems

Linux

### GGML backends

VirtGPU

### Problem description & steps to reproduce

Compiling with -DGGML_VIRTGPU on Fedora44/g++ 16.1.1 fails with `‘mutex’ in namespace ‘std’ does not name a type`


### First Bad Commit

015deb90485b9ebfff492e10fe7080a0439202b6

### Compile command

```shell
$ cmake -B build -DGGML_CCACHE=OFF -DGGML_RPC=ON -DCMAKE_INSTALL_PREFIX=/tmp/install -DLLAMA_BUILD_TESTS=OFF -DLLAMA_BUILD_EXAMPLES=OFF -DGGML_BUILD_TESTS=OFF -DGGML_BUILD_EXAMPLES=OFF -DGGML_NATIVE=OFF -DGGML_BACKEND_DL=ON -DGGML_CPU_ALL_VARIANTS=ON -DGGML_CMAKE_BUILD_TYPE=Release -DGGML_VIRTGPU=ON
...
cmake --build build --config Release -j12
```

### Relevant log output

```shell
/home/owalsh/llama.cpp/ggml/src/ggml-virtgpu/ggml-backend-device.cpp: In function ‘ggml_backend_buffer_type* ggml_backend_remoting_device_get_buffer_type(ggml_backend_dev_t)’:
/home/owalsh/llama.cpp/ggml/src/ggml-virtgpu/ggml-backend-device.cpp:80:21: error: ‘mutex’ in namespace ‘std’ does not name a type
   80 |         static std::mutex           mutex;
      |                     ^~~~~
/home/owalsh/llama.cpp/ggml/src/ggml-virtgpu/ggml-backend-device.cpp:2:1: note: ‘std::mutex’ is defined in header ‘<mutex>’; this is probably fixable by adding ‘#include <mutex>’
    1 | #include "ggml-remoting.h"
  +++ |+#include <mutex>
    2 | 
/home/owalsh/llama.cpp/ggml/src/ggml-virtgpu/ggml-backend-device.cpp:81:14: error: ‘lock_guard’ is not a member of ‘std’
   81 |         std::lock_guard<std::mutex> lock(mutex);
      |              ^~~~~~~~~~
/home/owalsh/llama.cpp/ggml/src/ggml-virtgpu/ggml-backend-device.cpp:81:14: note: ‘std::lock_guard’ is defined in header ‘<mutex>’; this is probably fixable by adding ‘#include <mutex>’
/home/owalsh/llama.cpp/ggml/src/ggml-virtgpu/ggml-backend-device.cpp:81:30: error: ‘mutex’ is not a member of ‘std’
   81 |         std::lock_guard<std::mutex> lock(mutex);
      |                              ^~~~~
/home/owalsh/llama.cpp/ggml/src/ggml-virtgpu/ggml-backend-device.cpp:81:30: note: ‘std::mutex’ is defined in header ‘<mutex>’; this is probably fixable by adding ‘#include <mutex>’
[ 20%] Linking CXX static library libcpp-httplib.a
/home/owalsh/llama.cpp/ggml/src/ggml-virtgpu/ggml-backend-device.cpp:81:42: error: ‘mutex’ was not declared in this scope
   81 |         std::lock_guard<std::mutex> lock(mutex);
      |                                          ^~~~~
/home/owalsh/llama.cpp/ggml/src/ggml-virtgpu/ggml-backend-device.cpp:81:37: error: ‘lock’ was not declared in this scope; did you mean ‘mlock’?
   81 |         std::lock_guard<std::mutex> lock(mutex);
      |                                     ^~~~
      |                                     mlock
/home/owalsh/llama.cpp/ggml/src/ggml-virtgpu/ggml-backend-device.cpp: In function ‘ggml_backend_buffer_type* ggml_backend_remoting_device_get_buffer_from_ptr_type(ggml_backend_dev_t)’:
/home/owalsh/llama.cpp/ggml/src/ggml-virtgpu/ggml-backend-device.cpp:103:21: error: ‘mutex’ in namespace ‘std’ does not name a type
  103 |         static std::mutex           mutex;
      |                     ^~~~~
/home/owalsh/llama.cpp/ggml/src/ggml-virtgpu/ggml-backend-device.cpp:103:16: note: ‘std::mutex’ is defined in header ‘<mutex>’; this is probably fixable by adding ‘#include <mutex>’
  103 |         static std::mutex           mutex;
      |                ^~~
/home/owalsh/llama.cpp/ggml/src/ggml-virtgpu/ggml-backend-device.cpp:104:14: error: ‘lock_guard’ is not a member of ‘std’
  104 |         std::lock_guard<std::mutex> lock(mutex);
      |              ^~~~~~~~~~
/home/owalsh/llama.cpp/ggml/src/ggml-virtgpu/ggml-backend-device.cpp:104:14: note: ‘std::lock_guard’ is defined in header ‘<mutex>’; this is probably fixable by adding ‘#include <mutex>’
/home/owalsh/llama.cpp/ggml/src/ggml-virtgpu/ggml-backend-device.cpp:104:30: error: ‘mutex’ is not a member of ‘std’
  104 |         std::lock_guard<std::mutex> lock(mutex);
      |                              ^~~~~
/home/owalsh/llama.cpp/ggml/src/ggml-virtgpu/ggml-backend-device.cpp:104:30: note: ‘std::mutex’ is defined in header ‘<mutex>’; this is probably fixable by adding ‘#include <mutex>’
/home/owalsh/llama.cpp/ggml/src/ggml-virtgpu/ggml-backend-device.cpp:104:42: error: ‘mutex’ was not declared in this scope
  104 |         std::lock_guard<std::mutex> lock(mutex);
      |                                          ^~~~~
/home/owalsh/llama.cpp/ggml/src/ggml-virtgpu/ggml-backend-device.cpp:104:37: error: ‘lock’ was not declared in this scope; did you mean ‘mlock’?
  104 |         std::lock_guard<std::mutex> lock(mutex);
      |                                     ^~~~
      |                                     mlock
gmake[2]: *** [ggml/src/ggml-virtgpu/CMakeFiles/ggml-virtgpu.dir/build.make:107: ggml/src/ggml-virtgpu/CMakeFiles/ggml-virtgpu.dir/ggml-backend-device.cpp.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:1836: ggml/src/ggml-virtgpu/CMakeFiles/ggml-virtgpu.dir/all] Error 2
```

--- a/ggml/src/ggml-virtgpu/ggml-backend-device.cpp
+++ b/ggml/src/ggml-virtgpu/ggml-backend-device.cpp
@@ -1,5 +1,7 @@
 #include "ggml-remoting.h"
 
+#include <mutex>
+
 static const char * ggml_backend_remoting_device_get_name(ggml_backend_dev_t dev) {
     virtgpu * gpu = DEV_TO_GPU(dev);
 

